### PR TITLE
Fix: fix pretty-printing of imports for static methods

### PR DESCRIPTION
**Pull request by danglotb**
Hello, 

I'm opening this pull-request that contains a test to reproduce the bug reported in #3618 and #3214.

Thank you

### CODE REVIEW

**Comment by monperrus** [comment]
Thanks @danglotb ! Are you trying to fix it?

**Comment by danglotb** [comment]
Hi @monperrus, I'll investigate how to. Does the test seem correct? Do you think I can use it as a base to fix the issue?

**Comment by danglotb** [comment]
Hi, @monperrus. Here my attempt of fixing these issues.

File: src/main/java/spoon/reflect/visitor/ImportCleaner.java [hunk 1]
```diff
@@ -85,7 +85,7 @@ protected void handleTargetedExpression(CtTargetedExpression<?, ?> targetedExpre
 			return;
 		}
 
-		if (target != null && target.isImplicit()) {
+		if (target.isImplicit()) {
```
Comment (monperrus) [inline src/main/java/spoon/reflect/visitor/ImportCleaner.java:88]: What about keeping `target != null` to avoid crashing with NPE on `target.isImplicit`?

**Review by monperrus — COMMENTED** [review]
Thanks a lot for the fix. LGTM, see mino comments.

Plus we have a no CI run, see https://github.com/INRIA/spoon/issues/3600

File: src/main/java/spoon/reflect/visitor/ImportCleaner.java (hunk 1 above)
Comment (danglotb) [reply src/main/java/spoon/reflect/visitor/ImportCleaner.java:88]: This check is redundant with the check just above([here](https://github.com/INRIA/spoon/blob/65e15161aacd2915b38e6b3ab8cd1d495a2c15be/src/main/java/spoon/reflect/visitor/ImportCleaner.java#L79)):
```java
if (target == null) {
   ...
   return;
}
```
If `target == null`, the condition `target.isImplicit()` will not be hit, isn't?

**Comment by monperrus** [comment]
OK for me. Unwip? Ready for merge?

**Comment by monperrus** [comment]
Thanks a lot @danglotb

**Pull request merged**